Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -733,52 +770,52 @@ extension HTTPParserError {
         switch fromCHTTPParserErrno {
-        case HPE_INVALID_EOF_STATE:
-            return .invalidEOFState
-        case HPE_HEADER_OVERFLOW:
-            return .headerOverflow
+        case HPE_INTERNAL:
+            return .invalidInternalState
+        case HPE_STRICT:
+            return .strictModeAssertion
+        case HPE_LF_EXPECTED:
+            return .lfExpected
+        case HPE_UNEXPECTED_CONTENT_LENGTH:
+            return .unexpectedContentLength
         case HPE_CLOSED_CONNECTION:
             return .closedConnection
-        case HPE_INVALID_VERSION:
-            return .invalidVersion
-        case HPE_INVALID_STATUS:
-            return .invalidStatus
         case HPE_INVALID_METHOD:
             return .invalidMethod
         case HPE_INVALID_URL:
             return .invalidURL
-        case HPE_INVALID_HOST:
-            return .invalidHost
-        case HPE_INVALID_PORT:
-            return .invalidPort
-        case HPE_INVALID_PATH:
-            return .invalidPath
-        case HPE_INVALID_QUERY_STRING:
-            return .invalidQueryString
-        case HPE_INVALID_FRAGMENT:
-            return .invalidFragment
-        case HPE_LF_EXPECTED:
-            return .lfExpected
-        case HPE_INVALID_HEADER_TOKEN:
+        case HPE_INVALID_CONSTANT:
+            return .invalidConstant
+        case HPE_INVALID_VERSION:
+            return .invalidVersion
+        case HPE_INVALID_HEADER_TOKEN,
+             HPE_UNEXPECTED_SPACE:
             return .invalidHeaderToken
         case HPE_INVALID_CONTENT_LENGTH:
             return .invalidContentLength
-        case HPE_UNEXPECTED_CONTENT_LENGTH:
-            return .unexpectedContentLength
         case HPE_INVALID_CHUNK_SIZE:
             return .invalidChunkSize
-        case HPE_INVALID_CONSTANT:
-            return .invalidConstant
-        case HPE_STRICT:
-            return .strictModeAssertion
-        case HPE_PAUSED:
+        case HPE_INVALID_STATUS:
+            return .invalidStatus
+        case HPE_INVALID_EOF_STATE:
+            return .invalidEOFState
+        case HPE_PAUSED, HPE_PAUSED_UPGRADE, HPE_PAUSED_H2_UPGRADE:
             return .paused
-        case HPE_UNKNOWN:
-            return .unknown
-        case HPE_INVALID_TRANSFER_ENCODING:
-            // The downside of enums here, we don't have a case for this. Map it to .unknown for now.
+        case HPE_INVALID_TRANSFER_ENCODING,
+             HPE_CR_EXPECTED,
+             HPE_CB_MESSAGE_BEGIN,
+             HPE_CB_HEADERS_COMPLETE,
+             HPE_CB_MESSAGE_COMPLETE,
+             HPE_CB_CHUNK_HEADER,
+             HPE_CB_CHUNK_COMPLETE,
+             HPE_USER,
+             HPE_CB_URL_COMPLETE,
+             HPE_CB_STATUS_COMPLETE,
+             HPE_CB_HEADER_FIELD_COMPLETE,
+             HPE_CB_HEADER_VALUE_COMPLETE:
+            // The downside of enums here, we don't have a case for these. Map them to .unknown for now.
```

Comment:
Bit annoying... I assume there's limited value in deprecating and replacing `HTTPParserError`?

Reply:
I don't see much value in it tbh.